// @expo-google-fonts/noto-sans@0.4.2/index.js
// This file is generated by @expo-google-fonts/generator
// If you want to modify it, go to packages/generator/templates

export * from './useFonts';
export { default as __metadata__ } from './metadata.json';
export const NotoSans_100Thin = require('./100Thin/NotoSans_100Thin.ttf');
export const NotoSans_200ExtraLight = require('./200ExtraLight/NotoSans_200ExtraLight.ttf');
export const NotoSans_300Light = require('./300Light/NotoSans_300Light.ttf');
export const NotoSans_400Regular = require('./400Regular/NotoSans_400Regular.ttf');
export const NotoSans_500Medium = require('./500Medium/NotoSans_500Medium.ttf');
export const NotoSans_600SemiBold = require('./600SemiBold/NotoSans_600SemiBold.ttf');
export const NotoSans_700Bold = require('./700Bold/NotoSans_700Bold.ttf');
export const NotoSans_800ExtraBold = require('./800ExtraBold/NotoSans_800ExtraBold.ttf');
export const NotoSans_900Black = require('./900Black/NotoSans_900Black.ttf');
export const NotoSans_100Thin_Italic = require('./100Thin_Italic/NotoSans_100Thin_Italic.ttf');
export const NotoSans_200ExtraLight_Italic = require('./200ExtraLight_Italic/NotoSans_200ExtraLight_Italic.ttf');
export const NotoSans_300Light_Italic = require('./300Light_Italic/NotoSans_300Light_Italic.ttf');
export const NotoSans_400Regular_Italic = require('./400Regular_Italic/NotoSans_400Regular_Italic.ttf');
export const NotoSans_500Medium_Italic = require('./500Medium_Italic/NotoSans_500Medium_Italic.ttf');
export const NotoSans_600SemiBold_Italic = require('./600SemiBold_Italic/NotoSans_600SemiBold_Italic.ttf');
export const NotoSans_700Bold_Italic = require('./700Bold_Italic/NotoSans_700Bold_Italic.ttf');
export const NotoSans_800ExtraBold_Italic = require('./800ExtraBold_Italic/NotoSans_800ExtraBold_Italic.ttf');
export const NotoSans_900Black_Italic = require('./900Black_Italic/NotoSans_900Black_Italic.ttf');
