// eas-cli@18.9.1/build/commands/observe/events.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const core_1 = require("@oclif/core");
const EasCommand_1 = tslib_1.__importDefault(require("../../commandUtils/EasCommand"));
const errors_1 = require("../../commandUtils/errors");
const flags_1 = require("../../commandUtils/flags");
const pagination_1 = require("../../commandUtils/pagination");
const generated_1 = require("../../graphql/generated");
const log_1 = tslib_1.__importDefault(require("../../log"));
const fetchEvents_1 = require("../../observe/fetchEvents");
const metricNames_1 = require("../../observe/metricNames");
const formatEvents_1 = require("../../observe/formatEvents");
const startAndEndTime_1 = require("../../observe/startAndEndTime");
const prompts_1 = require("../../prompts");
const json_1 = require("../../utils/json");
const DEFAULT_EVENTS_LIMIT = 10;
class ObserveEvents extends EasCommand_1.default {
    static hidden = true;
    static description = 'display individual app performance events ordered by metric value';
    static args = {
        metric: core_1.Args.string({
            description: 'Metric to query (e.g. tti, cold_launch)',
            required: false,
            options: Object.keys(metricNames_1.METRIC_ALIASES),
        }),
    };
    static flags = {
        sort: core_1.Flags.option({
            description: 'Sort order for events',
            options: Object.values(fetchEvents_1.EventsOrderPreset).map(s => s.toLowerCase()),
            required: false,
            default: fetchEvents_1.EventsOrderPreset.Oldest.valueOf().toLowerCase(),
        })(),
        platform: core_1.Flags.option({
            description: 'Filter by platform',
            options: Object.values(generated_1.AppObservePlatform).map(s => s.toLowerCase()),
        })(),
        after: core_1.Flags.string({
            description: 'Cursor for pagination. Use the endCursor from a previous query to fetch the next page.',
        }),
        limit: (0, pagination_1.getLimitFlagWithCustomValues)({
            defaultTo: DEFAULT_EVENTS_LIMIT,
            limit: 100,
        }),
        start: core_1.Flags.string({
            description: 'Start of time range (ISO date)',
            exclusive: ['days'],
        }),
        end: core_1.Flags.string({
            description: 'End of time range (ISO date)',
            exclusive: ['days'],
        }),
        days: core_1.Flags.integer({
            description: 'Show events from the last N days (mutually exclusive with --start/--end)',
            min: 1,
            exclusive: ['start', 'end'],
        }),
        'app-version': core_1.Flags.string({
            description: 'Filter by app version',
        }),
        'update-id': core_1.Flags.string({
            description: 'Filter by EAS update ID',
        }),
        'project-id': core_1.Flags.string({
            description: 'EAS project ID (defaults to the project ID of the current directory)',
        }),
        ...flags_1.EasNonInteractiveAndJsonFlags,
    };
    static contextDefinition = {
        ...this.ContextOptions.ProjectId,
        ...this.ContextOptions.LoggedIn,
    };
    static loggedInOnlyContextDefinition = {
        ...this.ContextOptions.LoggedIn,
    };
    async runAsync() {
        const { flags, args } = await this.parse(ObserveEvents);
        let projectId;
        let graphqlClient;
        if (flags['project-id']) {
            projectId = flags['project-id'];
            const ctx = await this.getContextAsync({ contextDefinition: ObserveEvents.loggedInOnlyContextDefinition }, { nonInteractive: flags['non-interactive'] });
            graphqlClient = ctx.loggedIn.graphqlClient;
        }
        else {
            const ctx = await this.getContextAsync(ObserveEvents, {
                nonInteractive: flags['non-interactive'],
            });
            projectId = ctx.projectId;
            graphqlClient = ctx.loggedIn.graphqlClient;
        }
        if (flags.json) {
            (0, json_1.enableJsonOutput)();
        }
        else {
            log_1.default.warn('EAS Observe is in preview and subject to breaking changes.');
        }
        let metricName;
        if (args.metric) {
            metricName = (0, metricNames_1.resolveMetricName)(args.metric);
        }
        else if (flags['non-interactive']) {
            throw new errors_1.EasCommandError('A metric argument is required in non-interactive mode. Available metrics: ' +
                Object.keys(metricNames_1.METRIC_ALIASES).join(', '));
        }
        else {
            const choices = Object.entries(metricNames_1.METRIC_SHORT_NAMES).map(([fullName, displayName]) => ({
                title: `${displayName} (${fullName})`,
                value: fullName,
            }));
            metricName = await (0, prompts_1.selectAsync)('Select a metric', choices);
        }
        const orderBy = (0, fetchEvents_1.resolveOrderBy)(flags.sort);
        const { daysBack, startTime, endTime } = (0, startAndEndTime_1.resolveTimeRange)(flags);
        const platform = flags.platform
            ? flags.platform === 'android'
                ? generated_1.AppObservePlatform.Android
                : generated_1.AppObservePlatform.Ios
            : undefined;
        const platforms = platform
            ? [platform === generated_1.AppObservePlatform.Android ? generated_1.AppPlatform.Android : generated_1.AppPlatform.Ios]
            : [generated_1.AppPlatform.Android, generated_1.AppPlatform.Ios];
        const [{ events, pageInfo }, totalEventCount] = await Promise.all([
            (0, fetchEvents_1.fetchObserveEventsAsync)(graphqlClient, projectId, {
                metricName,
                orderBy,
                limit: flags.limit ?? DEFAULT_EVENTS_LIMIT,
                ...(flags.after && { after: flags.after }),
                startTime,
                endTime,
                platform,
                appVersion: flags['app-version'],
                updateId: flags['update-id'],
            }),
            (0, fetchEvents_1.fetchTotalEventCountAsync)(graphqlClient, projectId, metricName, platforms, startTime, endTime),
        ]);
        if (flags.json) {
            (0, json_1.printJsonOnlyOutput)((0, formatEvents_1.buildObserveEventsJson)(events, pageInfo));
        }
        else {
            log_1.default.addNewLineIfNone();
            log_1.default.log((0, formatEvents_1.buildObserveEventsTable)(events, pageInfo, {
                metricName,
                daysBack,
                startTime,
                endTime,
                totalEventCount,
            }));
        }
    }
}
exports.default = ObserveEvents;
