// @expo-google-fonts/source-sans-3@0.4.1/index.js
// This file is generated by @expo-google-fonts/generator
// If you want to modify it, go to packages/generator/templates

export * from './useFonts';
export { default as __metadata__ } from './metadata.json';
export const SourceSans3_200ExtraLight = require('./200ExtraLight/SourceSans3_200ExtraLight.ttf');
export const SourceSans3_300Light = require('./300Light/SourceSans3_300Light.ttf');
export const SourceSans3_400Regular = require('./400Regular/SourceSans3_400Regular.ttf');
export const SourceSans3_500Medium = require('./500Medium/SourceSans3_500Medium.ttf');
export const SourceSans3_600SemiBold = require('./600SemiBold/SourceSans3_600SemiBold.ttf');
export const SourceSans3_700Bold = require('./700Bold/SourceSans3_700Bold.ttf');
export const SourceSans3_800ExtraBold = require('./800ExtraBold/SourceSans3_800ExtraBold.ttf');
export const SourceSans3_900Black = require('./900Black/SourceSans3_900Black.ttf');
export const SourceSans3_200ExtraLight_Italic = require('./200ExtraLight_Italic/SourceSans3_200ExtraLight_Italic.ttf');
export const SourceSans3_300Light_Italic = require('./300Light_Italic/SourceSans3_300Light_Italic.ttf');
export const SourceSans3_400Regular_Italic = require('./400Regular_Italic/SourceSans3_400Regular_Italic.ttf');
export const SourceSans3_500Medium_Italic = require('./500Medium_Italic/SourceSans3_500Medium_Italic.ttf');
export const SourceSans3_600SemiBold_Italic = require('./600SemiBold_Italic/SourceSans3_600SemiBold_Italic.ttf');
export const SourceSans3_700Bold_Italic = require('./700Bold_Italic/SourceSans3_700Bold_Italic.ttf');
export const SourceSans3_800ExtraBold_Italic = require('./800ExtraBold_Italic/SourceSans3_800ExtraBold_Italic.ttf');
export const SourceSans3_900Black_Italic = require('./900Black_Italic/SourceSans3_900Black_Italic.ttf');
