// eas-cli@18.9.1/build/graphql/queries/ObserveQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObserveQuery = void 0;
const tslib_1 = require("tslib");
const graphql_tag_1 = tslib_1.__importDefault(require("graphql-tag"));
const client_1 = require("../client");
const graphql_1 = require("graphql");
const Observe_1 = require("../types/Observe");
exports.ObserveQuery = {
    async timeSeriesAsync(graphqlClient, { appId, metricName, platform, startTime, endTime, }) {
        const data = await (0, client_1.withErrorHandlingAsync)(graphqlClient
            .query((0, graphql_tag_1.default) `
            query AppObserveTimeSeries(
              $appId: String!
              $input: AppObserveTimeSeriesInput!
            ) {
              app {
                byId(appId: $appId) {
                  id
                  observe {
                    timeSeries(input: $input) {
                      ...AppObserveTimeSeriesFragment
                    }
                  }
                }
              }
            }
            ${(0, graphql_1.print)(Observe_1.AppObserveAppVersionFragmentNode)}
            ${(0, graphql_1.print)(Observe_1.AppObserveTimeSeriesFragmentNode)}
          `, {
            appId,
            input: { metricName, platform, startTime, endTime },
        })
            .toPromise());
        return data.app.byId.observe.timeSeries;
    },
    async appVersionsAsync(graphqlClient, { appId, platform, startTime, endTime, metricNames, }) {
        const data = await (0, client_1.withErrorHandlingAsync)(graphqlClient
            .query((0, graphql_tag_1.default) `
            query AppObserveAppVersions(
              $appId: String!
              $input: AppObserveReleasesInput!
            ) {
              app {
                byId(appId: $appId) {
                  id
                  observe {
                    appVersions(input: $input) {
                      ...AppObserveAppVersionFragment
                    }
                  }
                }
              }
            }
            ${(0, graphql_1.print)(Observe_1.AppObserveAppVersionFragmentNode)}
          `, {
            appId,
            input: { platform, startTime, endTime, ...(metricNames && { metricNames }) },
        })
            .toPromise());
        return data.app.byId.observe.appVersions;
    },
    async eventsAsync(graphqlClient, variables) {
        const data = await (0, client_1.withErrorHandlingAsync)(graphqlClient
            .query((0, graphql_tag_1.default) `
            query AppObserveEvents(
              $appId: String!
              $filter: AppObserveEventsFilter
              $first: Int
              $after: String
              $orderBy: AppObserveEventsOrderBy
            ) {
              app {
                byId(appId: $appId) {
                  id
                  observe {
                    events(
                      filter: $filter
                      first: $first
                      after: $after
                      orderBy: $orderBy
                    ) {
                      pageInfo {
                        hasNextPage
                        hasPreviousPage
                        endCursor
                      }
                      edges {
                        cursor
                        node {
                          id
                          ...AppObserveEventFragment
                        }
                      }
                    }
                  }
                }
              }
            }
            ${(0, graphql_1.print)(Observe_1.AppObserveEventFragmentNode)}
          `, variables)
            .toPromise());
        const { edges, pageInfo } = data.app.byId.observe.events;
        return {
            events: edges.map(edge => edge.node),
            pageInfo,
        };
    },
};
