// eas-cli@18.9.1/build/commands/simulator/stop.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const core_1 = require("@oclif/core");
const EasCommand_1 = tslib_1.__importDefault(require("../../commandUtils/EasCommand"));
const flags_1 = require("../../commandUtils/flags");
const DeviceRunSessionMutation_1 = require("../../graphql/mutations/DeviceRunSessionMutation");
const ora_1 = require("../../ora");
class SimulatorStop extends EasCommand_1.default {
    static hidden = true;
    static description = '[EXPERIMENTAL] stop a remote simulator session on EAS by its device run session ID';
    static flags = {
        id: core_1.Flags.string({
            description: 'Device run session ID',
            required: true,
        }),
        ...flags_1.EASNonInteractiveFlag,
    };
    static contextDefinition = {
        ...this.ContextOptions.LoggedIn,
    };
    async runAsync() {
        const { flags } = await this.parse(SimulatorStop);
        const { loggedIn: { graphqlClient }, } = await this.getContextAsync(SimulatorStop, {
            nonInteractive: flags['non-interactive'],
        });
        const stopSpinner = (0, ora_1.ora)(`🛑 Stopping device run session ${flags.id}`).start();
        try {
            const session = await DeviceRunSessionMutation_1.DeviceRunSessionMutation.stopDeviceRunSessionAsync(graphqlClient, flags.id);
            stopSpinner.succeed(`🎉 Device run session ${session.id} is ${session.status.toLowerCase()}`);
        }
        catch (err) {
            stopSpinner.fail(`Failed to stop device run session ${flags.id}`);
            throw err;
        }
    }
}
exports.default = SimulatorStop;
