// @expo-google-fonts/crimson-pro@0.4.2/index.js
// This file is generated by @expo-google-fonts/generator
// If you want to modify it, go to packages/generator/templates

export * from './useFonts';
export { default as __metadata__ } from './metadata.json';
export const CrimsonPro_200ExtraLight = require('./200ExtraLight/CrimsonPro_200ExtraLight.ttf');
export const CrimsonPro_300Light = require('./300Light/CrimsonPro_300Light.ttf');
export const CrimsonPro_400Regular = require('./400Regular/CrimsonPro_400Regular.ttf');
export const CrimsonPro_500Medium = require('./500Medium/CrimsonPro_500Medium.ttf');
export const CrimsonPro_600SemiBold = require('./600SemiBold/CrimsonPro_600SemiBold.ttf');
export const CrimsonPro_700Bold = require('./700Bold/CrimsonPro_700Bold.ttf');
export const CrimsonPro_800ExtraBold = require('./800ExtraBold/CrimsonPro_800ExtraBold.ttf');
export const CrimsonPro_900Black = require('./900Black/CrimsonPro_900Black.ttf');
export const CrimsonPro_200ExtraLight_Italic = require('./200ExtraLight_Italic/CrimsonPro_200ExtraLight_Italic.ttf');
export const CrimsonPro_300Light_Italic = require('./300Light_Italic/CrimsonPro_300Light_Italic.ttf');
export const CrimsonPro_400Regular_Italic = require('./400Regular_Italic/CrimsonPro_400Regular_Italic.ttf');
export const CrimsonPro_500Medium_Italic = require('./500Medium_Italic/CrimsonPro_500Medium_Italic.ttf');
export const CrimsonPro_600SemiBold_Italic = require('./600SemiBold_Italic/CrimsonPro_600SemiBold_Italic.ttf');
export const CrimsonPro_700Bold_Italic = require('./700Bold_Italic/CrimsonPro_700Bold_Italic.ttf');
export const CrimsonPro_800ExtraBold_Italic = require('./800ExtraBold_Italic/CrimsonPro_800ExtraBold_Italic.ttf');
export const CrimsonPro_900Black_Italic = require('./900Black_Italic/CrimsonPro_900Black_Italic.ttf');
