// @expo-google-fonts/eb-garamond@0.4.2/index.js
// This file is generated by @expo-google-fonts/generator
// If you want to modify it, go to packages/generator/templates

export * from './useFonts';
export { default as __metadata__ } from './metadata.json';
export const EBGaramond_400Regular = require('./400Regular/EBGaramond_400Regular.ttf');
export const EBGaramond_500Medium = require('./500Medium/EBGaramond_500Medium.ttf');
export const EBGaramond_600SemiBold = require('./600SemiBold/EBGaramond_600SemiBold.ttf');
export const EBGaramond_700Bold = require('./700Bold/EBGaramond_700Bold.ttf');
export const EBGaramond_800ExtraBold = require('./800ExtraBold/EBGaramond_800ExtraBold.ttf');
export const EBGaramond_400Regular_Italic = require('./400Regular_Italic/EBGaramond_400Regular_Italic.ttf');
export const EBGaramond_500Medium_Italic = require('./500Medium_Italic/EBGaramond_500Medium_Italic.ttf');
export const EBGaramond_600SemiBold_Italic = require('./600SemiBold_Italic/EBGaramond_600SemiBold_Italic.ttf');
export const EBGaramond_700Bold_Italic = require('./700Bold_Italic/EBGaramond_700Bold_Italic.ttf');
export const EBGaramond_800ExtraBold_Italic = require('./800ExtraBold_Italic/EBGaramond_800ExtraBold_Italic.ttf');
